Document "How to Connect to Asset Hub" (#365)
polkadot-developers/polkadot-docs: develop/smart-contracts/connect-to-asset-hub.md

---
title: Connect to Asset Hub
description: Explore how to connect to different Asset Hub networks, configure your wallet, and obtain test tokens for developing and testing smart contracts.
---

> # Connect to Asset Hub

Reviewer: Well this method works for any Ethereum Compatible wallet that supports https://eips.ethereum.org/EIPS/eip-3085

I believe this is compatible with many wallets, not only Metamask

> # Connect to Asset Hub

Author: Oh, I didn't know that. Okay, that makes more sense now. Let me move it to the top

> # Connect to Asset Hub

Author: done [e49349e](https://github.com/polkadot-developers/polkadot-docs/pull/365/commits/e49349ef8a26e8c747db9c1dc985c111d12c3b7b)

## Networks

Developers can leverage Asset Hub across diverse networks, from TestNets to MainNet. This section outlines the network specifications and connection details for each environment.

> 

Reviewer 2: I would add one more subsection to be future-proof and be prepared when Kusama and Polkadot AssetHub are ready.
Something like:
```
## Networks
### Westend Asset Hub
####Specifications
#### Endpoints
```

Author: I disagree here. If we do that, we will end up with a very long page once Kusama ah, Paseo ah, and Polkadot ah are ready. Also, as far as I know, the 4th `####` is not being used in our documentation according to the writing guidelines.

I used the tabs to add the future networks in the following way:

![image](https://github.com/user-attachments/assets/d134dac1-83b4-460f-b4f1-6896273c02c2)

Reviewer 2: The tabs look way better. Let's keep using that

### Specifications

> === "Westend Asset Hub"

Reviewer: Add the HTTPs endpoint (at least one) so that developers can just easily copy paste

Author: fixed in [f4c503c](https://github.com/polkadot-developers/polkadot-docs/pull/365/commits/f4c503c26588028d444a54f64cc7be6e1adf307d)

    Network name
    ```
    Asset-Hub Westend TestNet
    ```

    ---

    Currency symbol
    ```
    WND
    ```

    ---
    
    Chain ID
    ```
    420420421
    ```

    ---
    
    Block explorer URL
    ```
    https://assethub-westend.subscan.io
    ```

### Endpoints

> === "Westend Asset Hub"

Reviewer: Change this for HTTPs endpoitns as those are the ones more broadly accepted by wallets

Author: fixed in [f4c503c](https://github.com/polkadot-developers/polkadot-docs/pull/365/commits/f4c503c26588028d444a54f64cc7be6e1adf307d)

    ```

## Connect your Wallet

### Metamask

MetaMask is a popular wallet for interacting with EVM-compatible chains. It allows users to connect to test networks that support Ethereum-based smart contracts. However, it's important to emphasize that MetaMask primarily facilitates interactions with smart contracts, giving users access to various chain functionalities.

To get started with MetaMask, you need to install the [MetaMask extension](https://metamask.io/download/){target=\_blank} and add it to the browser. Once you install MetaMask, you can set up a new wallet and securely store your seed phrase. This phrase is crucial for recovery in case you lose access.

For example, to connect to the Westend Asset Hub TestNet via MetaMask, you need to follow these steps:

1. Open the MetaMask extension and click on the network icon to switch to the Asset Hub Westend TestNet

    ![](/images/develop/smart-contracts/connect-to-asset-hub/connect-to-asset-hub-1.webp){: .browser-extension}

2. Click on the **Add a custom network** button

    ![](/images/develop/smart-contracts/connect-to-asset-hub/connect-to-asset-hub-2.webp){: .browser-extension}

> 3. Complete the necessary fields using the specified parameters, then click the **Save** button (refer to the [Networks](#networks) section for copy and paste parameters)

Reviewer 2: ```suggestion
3. Complete the necessary fields using the parameters in the [Specifications](#specifications) section, then click the **Save** button (refer to the [Networks](#networks) section for copy and paste parameters)
```

    ![](/images/develop/smart-contracts/connect-to-asset-hub/connect-to-asset-hub-3.webp){: .browser-extension}

4. Click on **Asset-Hub Westend TestNet** to switch the network

    ![](/images/develop/smart-contracts/connect-to-asset-hub/connect-to-asset-hub-4.webp){: .browser-extension}

> The same steps as above can be used to connect to any Asset Hub chain, modifying the corresponding parameters.

Reviewer 3: ```suggestion
The steps in the preceding section can be used to connect to any Asset Hub chain by modifying the network specification and endpoint parameters.
```

## Test Tokens

> To perform transactions and engage with smart contracts on any Asset Hub chain, you will require tokens. Here's how to obtain Westend Asset Hub (WND) tokens for testing purposes:

Reviewer 3: ```suggestion
You will need testnet tokens to perform transactions and engage with smart contracts on any Asset Hub chain. Here's how to obtain Westend Asset Hub (WND) tokens for testing purposes:
```

> 1. Navigate to the [Polkadot Faucet](https://faucet.polkadot.io){target=\_blank}

Reviewer 3: ```suggestion
1. Navigate to the [Polkadot Faucet](https://faucet.polkadot.io){target=\_blank}
```
When I click this link, it opens to Paseo. Please add something like "If not already selected..." telling them to use the Network drop-down so the user knows what to do if they see the same thing.

Author: fixed in [ad5ccba](https://github.com/polkadot-developers/polkadot-docs/pull/365/commits/ad5ccbab85ebfe5448f72c4a1e919a4f660e9dad)

> 2. Copy your MetaMask address linked to the Westend Asset Hub and paste it into the designated field

Reviewer 2: ```suggestion
2. Copy your MetaMask address linked to Westend Asset Hub and paste it into the designated field
```

    <a href="#" class="md-button connectMetaMask" value="westendAssetHub">Connect to Westend Asset Hub</a>
</div>

## Networks

Developers can leverage Asset Hub across diverse networks, from TestNets to MainNet. This section outlines the network specifications and connection details for each environment.

### Specifications

=== "Westend Asset Hub"

    Network name
    ```
    Asset-Hub Westend TestNet
    ```

    ---

    Currency symbol
    ```
    WND
    ```

    ---
    
    Chain ID
    ```
    420420421
    ```

    ---
    
    RPC URL
    ```
    https://westend-asset-hub-eth-rpc.polkadot.io
    ```

    ---
    
    Block explorer URL
    ```
    https://assethub-westend.subscan.io
    ```

### Endpoints

=== "Westend Asset Hub"
    Dwellir
    ```
    https://westend-asset-hub-eth-rpc.polkadot.io
    ```
    ---
    Dwellir Tunisia
    ```
    https://westmint-rpc.polkadot.io
    ```
    ---
    IBP1
    ```
    https://rpc.ibp.network/asset-hub-westend
    ```
    ---
    IBP2
    ```
    https://asset-hub-westend.dotters.network
    ```
    ---
    Parity
    ```
    https://westend-asset-hub-rpc.polkadot.io
    ```

## Connect your Wallet

### Metamask

> MetaMask is a popular wallet for interacting with EVM-compatible chains. It allows users to connect to test networks that support Ethereum-based smart contracts. However, it's important to emphasize that MetaMask primarily facilitates interactions with smart contracts, giving users access to various chain functionalities. 

Reviewer 4: ```suggestion
[MetaMask](https://metamask.io/){target=\_blank} is a popular wallet for interacting with EVM-compatible chains. It allows users to connect to test networks that support Ethereum-based smart contracts. However, it's important to emphasize that MetaMask primarily facilitates interactions with smart contracts, giving users access to various chain functionalities. 
```

First mention of this we should have a link to their website

    ![](/images/develop/smart-contracts/connect-to-asset-hub/connect-to-asset-hub-1.webp){: .browser-extension}

2. Click on the **Add a custom network** button

    ![](/images/develop/smart-contracts/connect-to-asset-hub/connect-to-asset-hub-2.webp){: .browser-extension}

3. Complete the necessary fields using the parameters in the [Specifications](#specifications) section, then click the **Save** button (refer to the [Networks](#networks) section for copy and paste parameters)

    ![](/images/develop/smart-contracts/connect-to-asset-hub/connect-to-asset-hub-3.webp){: .browser-extension}

4. Click on **Asset-Hub Westend TestNet** to switch the network

    ![](/images/develop/smart-contracts/connect-to-asset-hub/connect-to-asset-hub-4.webp){: .browser-extension}

The steps in the preceding section can be used to connect to any Asset Hub chain by modifying the network specification and endpoint parameters.

## Test Tokens

You will need testnet tokens to perform transactions and engage with smart contracts on any Asset Hub chain. Here's how to obtain Westend Asset Hub (WND) tokens for testing purposes:

> 1. Navigate to the [Polkadot Faucet](https://faucet.polkadot.io){target=\_blank}. If the desired network is not already selected, choose it from the Network drop-down.

Reviewer 4: ```suggestion
1. Navigate to the [Polkadot Faucet](https://faucet.polkadot.io){target=\_blank}. If the desired network is not already selected, choose it from the Network drop-down
```
